scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -54,74 +53,54 @@ def load_light_scenarios_from_jsonl(path: str) -> List[LightScenario]:
-    {
-        "light_scenario_key": {
-            "metadata":{
-                "split": "SPLIT",
-                "scenario_attribute_1": "ATTRIBUTE1",
-                "scenario_attribute_2": "ATTRIBUTE2",
-            }
-        },
-        "light_instances": [
-            {
-                "input": "INPUT_TEXT1",
-                "references": [
-                    "REFERENCE_TEXT_1",
-                    "REFERENCE_TEXT_2"
-                ]
-            },
-            {
-                "input": "INPUT_TEXT2",
-                "references": [
-                    "REFERENCE_TEXT_3",
-                    "REFERENCE_TEXT_4"
-                ]
-            }
-        ]
-    }
-
-    Note that the values of light_scenario_key.metadata need to be hashable.
     """
 
     def create_light_instance_from_dict(instance_dict: dict) -> LightInstance:
-        return LightInstance(input=instance_dict["input"], references=instance_dict["references"])
+        return LightInstance(
+            input=instance_dict[PART_INPUT], references=instance_dict[PART_REF], id=instance_dict["id"]
+        )
 
     light_scenarios: List[LightScenario] = []
     light_scenario_jsons = open(path, "r").readlines()
     for light_scenario_json in light_scenario_jsons:
         light_scenario_dict: dict = json.loads(light_scenario_json)
 
-        light_scenario_metadata: dict = light_scenario_dict["light_scenario_key"]["metadata"]
+        light_scenario_key_dict: dict = light_scenario_dict["scenario_key"]
         # if the light_scenarios are exported from helm, they will have a scenario_spec field
-        if "scenario_spec" in light_scenario_metadata:
-            light_scenario_metadata["scenario_spec"] = ScenarioSpec(**light_scenario_metadata["scenario_spec"])
-        light_scenario_key = LightScenarioKey(metadata=light_scenario_metadata)
+        scenario_spec = ScenarioSpec(**light_scenario_key_dict["scenario_spec"])
+        light_scenario_key = LightScenarioKey(scenario_spec=scenario_spec, split=light_scenario_key_dict["split"])
         light_instances: List[LightInstance] = [
-            create_light_instance_from_dict(instance_dict) for instance_dict in light_scenario_dict["light_instances"]
+            create_light_instance_from_dict(instance_dict) for instance_dict in light_scenario_dict["instances"]
         ]
-        light_scenarios.append(LightScenario(light_scenario_key=light_scenario_key, light_instances=light_instances))
+        light_scenarios.append(LightScenario(scenario_key=light_scenario_key, instances=light_instances))
     return light_scenarios
 
 
 def create_ngram_index(
-    light_scenarios: List[LightScenario], n_values: List[int], tokenizer: LightTokenizer
+    light_scenarios: List[LightScenario],
+    n_values: List[int],
+    tokenizer: LightTokenizer,
+    stats_key_counts: Dict[DataOverlapStatsKey, int],
 ) -> NgramIndex:
     """Given a list of scenarios and n values, initialize ngram_index"""
```

Comment:
Update docstring to explain that `stats_key_counts` is an output variable (or better yet, just return `Tuple[NgramIndex,Dict[DataOverlapStatsKey, int]]`)

Reply:
Updated